### PR TITLE
Remove the min_compatible_shard_node option and associated classes

### PR DESCRIPTION
**Pull request by thecoop**
Now that SqlVersion is separated from Version, remove use of `min_compatible_shard_node` search option and retry functionality. If this functionality is needed in the future, it should instead use capabilities.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/search/VersionMismatchException.java [hunk 1]
```diff
@@ -1,27 +0,0 @@
...
- */
-
-package org.elasticsearch.action.search;
-
-import org.elasticsearch.ElasticsearchException;
-import org.elasticsearch.common.io.stream.StreamInput;
-
-import java.io.IOException;
-
-public class VersionMismatchException extends ElasticsearchException {
```
Comment (thecoop) [inline server/src/main/java/org/elasticsearch/action/search/VersionMismatchException.java:17]: I'm not sure it's worth keeping this class and its registrations around, for if/when we do need capabilities checks on search queries?

Comment (ldematte) [reply server/src/main/java/org/elasticsearch/action/search/VersionMismatchException.java:17]: I'd say delete it; if/when we need new checks we can introduce a new exception.

**Review by ldematte — APPROVED** [review]
From a Core/Infra point of view, all OK, but you probably want to wait for input/approval from analytics

**Review by bpintea — APPROVED** [review]
Lgtm

**Pull request merged**